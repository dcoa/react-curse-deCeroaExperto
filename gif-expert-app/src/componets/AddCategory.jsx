import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({ setCategory }) => {
    const [inputValue, setinputValue] = useState('');

    const handleSubmit = (e) => {
        // eslint-disable-next-line no-unused-expressions
        e.preventDefault();
        if( inputValue.trim().length > 2){
            setCategory(categ => [inputValue, ...categ])
            setinputValue('')
        }
    }
    return (
        <>
            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    value={inputValue}
                    placeholder='Escribe lo que quieres buscar y presiona "Enter"'
                    onChange={e => setinputValue(e.target.value)}
                />
            </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}
