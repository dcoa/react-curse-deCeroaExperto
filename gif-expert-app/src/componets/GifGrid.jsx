import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifts } from '../hooks/useFetchGifts';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ Category }) => {
    // const [images, setImages] = useState([]);
    const { data, loading } = useFetchGifts(Category);
    //transformaci√≥n para crear una galeria de imagenes
    let columns = [];

    for (let i = 0; i < data.length; i += 3) {
        columns.push(data.slice(i, i + 3))
    }

    return (
        <>
            <h3>{Category}</h3>
            <div className='card-grid'>
                {loading && 'Cargandoo...'}
                {
                    columns.map((images, i) => <div key={'columns' + i} className='columns'>
                        {images.map(img => <GifGridItem key={img.id} {...img} />)}
                    </div>)
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    Category: PropTypes.string.isRequired
}
