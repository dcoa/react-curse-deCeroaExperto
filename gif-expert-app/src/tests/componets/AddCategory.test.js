import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from './../../componets/AddCategory'

describe('Test Add Category component', () => {
    const setstate = jest.fn();
    let wrapper;
    
    beforeEach( () => {
        wrapper = shallow(<AddCategory setCategory={setstate}/>)
    });

    test('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should change the input text', () => {
        const input = wrapper.find('input');
        // es necesario enviar el evento que tiene forma de objeto
        const value = 'Hola mundo';
        input.simulate('change', { target: { value }});
        // En este caso no va a encontrar el valor porque la funcion se ejecuta y limpia el valor
        //expect(input.prop('value')).toBe(value);
    })

    test('should no post the information onSubmit', () => {

        // SimulaciÃ³n del envio del formulario, preventDefault forma corta para definir la funcion dentro del objeto
        wrapper.find('form').simulate('submit', {preventDefault(){}})
        // La funcion setstate no se debe haber llammado.
        expect(setstate).not.toHaveBeenCalled()
    })
    test('should call setstate and clear the input text', () => {
        const input = wrapper.find('input');
        input.simulate('change', { target: { value:'hisoka' }});
        wrapper.find('form').simulate('submit', {preventDefault(){}})
        expect(setstate).toHaveBeenCalled()
        expect(input.prop('value')).toBe('')
        // se espera que el setCategory se llame con una funcion como paramentro
        expect( setstate ).toHaveBeenCalledWith(expect.any(Function))
    })
    
});
