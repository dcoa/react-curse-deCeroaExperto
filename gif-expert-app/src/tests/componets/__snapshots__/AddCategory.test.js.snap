// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Add Category component should render correct 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Escribe lo que quieres buscar y presiona \\"Enter\\""
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
